### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by gabyang**
Please have mercy, took me a while to figure this out

### CODE REVIEW

File: Auth/AuthAPI.py [hunk 1]
```diff
@@ -1,7 +1,52 @@
 # Score API here
 from flask import Blueprint
+from flask import request
+from flask import jwt
```
Comment (nobelang78) [inline Auth/AuthAPI.py:4]: Flask does not have jwt, just import jwt. We are using the package PyJWT which has the name signature jwt

File: main.py [hunk 2]
```diff
@@ -10,3 +10,13 @@
...
+
+@app.route('/', methods = ["GET"])
+def main():
+    return "Welcome User"
+
+# In main: GET / homepage that returns a welcome message
+# 
+
+
+# Write your flask code here
```
Comment (nobelang78) [inline main.py:22]: No code to start the server, look into app.run

File: Profiles/ProfilesAPI.py [hunk 3]
```diff
@@ -1,7 +1,57 @@
...
 profiles_api = Blueprint("profiles", __name__)
+
+@profiles_api.route("/<string:id>", methods = ["GET"])
+def searchProfiles(id):
+    for set in db:
+        if id == set["name"]:
+            return set
+    return "Wrong id" #What type of error should I throw in here
+
+@profiles_api.route("/profiles", methods = ["POST"])
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:18]: /profiles already added as a prefix, this results in "/profiles/profiles" needed

File: Profiles/ProfilesAPI.py [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
 sys.path.append("../")
 
 profiles_api = Blueprint("profiles", __name__)
+
+@profiles_api.route("/<string:id>", methods = ["GET"])
+def searchProfiles(id):
+    for set in db:
+        if id == set["name"]:
+            return set
+    return "Wrong id" #What type of error should I throw in here
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:16]: Should return a 404 not found error

File: Profiles/ProfilesAPI.py [hunk 5]
```diff
@@ -1,7 +1,57 @@
...
+    db.append
+    ({
+        "name": request.form["name"],
+        "score": [] #empty because it creates name only 
+    })
+    return "Profile created"
+
+@profiles_api.route('/<string:id>', methods = ["DELETE"])
+def deleteProfile(id):
+    flash("Are you sure you want to delete profile?")
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:33]: Flashing is not used. Flashing is to carry messages between request. RESTFUL APis do not do that. We simply return the data + message that the user requested

File: Profiles/ProfilesAPI.py [hunk 6]
```diff
@@ -1,7 +1,57 @@
...
+@profiles_api.route('/<string:id>/score', methods = ["GET"])
+def profileScore(id):
+    allScores = []
+    for set in db:
+        if id == set["name"]:
+            allScores = set["scores"]
+    if (allScores == []):
+        return "No scores are found"    
+    aboveMinScore = list(filter(lambda x: x > int(request.args.get("minScore"))), allScores)
+    return aboveMinScore
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:52]: list cannot be returned, wrap in dict/jsonify

File: Auth/AuthAPI.py [hunk 7]
```diff
@@ -1,7 +1,52 @@
...
 import sys
 from db import db
 sys.path.append("../")
 
 auth_api = Blueprint("auth", __name__)
+
+temporaryLocalArr = []
+@auth_api.route('/register', methods = ["POST"])
+def register():
+    if (request.args.get("username") == None or request.args.get("hashedPassword") == None):
```
Comment (nobelang78) [inline Auth/AuthAPI.py:16]: should be fromr request.form for form data in POST body

File: Auth/AuthAPI.py [hunk 8]
```diff
@@ -1,7 +1,52 @@
...
+            return "login successful"
+        return "login unsuccessful"
+    
+    user = {
+    "username": request.args.get("username"),
+    "hashedPassword": request.args.get("hashedPassword")
+    }
+    if user in temporaryLocalArr:
+        newToken = jwt.encode(user, current_app.config['SECRET_KEY'], algorithm="HS256")
+        flash("login succesful")
```
Comment (nobelang78) [inline Auth/AuthAPI.py:39]: can just reutrn login successful + a token in dict

File: Auth/AuthAPI.py [hunk 9]
```diff
@@ -1,7 +1,52 @@
...
+    })
+    return "successfully registered"
+
+@auth_api.route('/login', methods = ["POST"])
+def login():
+    token = request.args.get("token")
+    if token != None:
+        data = jwt.decode(token, current_app.config['SECRET_KEY'], algorithms=["HS256"])
+        if data in temporaryLocalArr:
+            return "login successful"
```
Comment (nobelang78) [inline Auth/AuthAPI.py:30]: Should be 403 forbidden

**Review by codecodecoder78 — CHANGES_REQUESTED** [review]

File: Auth/AuthAPI.py (hunk 9 above)
Comment (gabyang) [reply Auth/AuthAPI.py:30]: sorry, not so clear on this, what is forbidden? Do you mean the return string "Login Successful" should be "403 Forbidden"

File: Auth/AuthAPI.py (hunk 1 above)
Comment (gabyang) [reply Auth/AuthAPI.py:4]: done

File: Profiles/ProfilesAPI.py (hunk 3 above)
Comment (gabyang) [reply Profiles/ProfilesAPI.py:18]: done

File: Profiles/ProfilesAPI.py (hunk 4 above)
Comment (gabyang) [reply Profiles/ProfilesAPI.py:16]: done

File: Profiles/ProfilesAPI.py (hunk 5 above)
Comment (gabyang) [reply Profiles/ProfilesAPI.py:33]: done

File: Auth/AuthAPI.py (hunk 9 above)
Comment (nobelang78) [reply Auth/AuthAPI.py:30]: Sorry, referring to the unsuccessful case. As this is a failed login, the status code for the return should be 403

**Pull request closed without merging**